### PR TITLE
Removes deprecated use and raw types

### PR DESCRIPTION
**Pull request by zregvart**
This cleans up the Netty implementation by replacing deprecated usage
and adds type parameters to raw types used.

### CODE REVIEW

**Comment by zregvart** [comment]
I'm not having any luck with the CI build.

**Comment by jbertram** [comment]
I ran the PR build locally (i.e. 'mvn -Pfast-tests -Pextra-tests test'). It's hanging on NettyConnectorTest.testStartStop which works on the master branch. It looks like your change broke something.

File: artemis-core-client/src/main/java/org/apache/activemq/artemis/core/remoting/impl/netty/SharedEventLoopGroup.java [hunk 1]
```diff
@@ -47,15 +47,19 @@ private SharedEventLoopGroup(EventLoopGroup eventLoopGroup) {
 
    public static synchronized void forceShutdown() {
       if (instance != null) {
-         instance.shutdown();
+         try {
+            instance.shutdownGracefully(100, 3000, TimeUnit.MILLISECONDS).await();
```
Comment (zregvart) [inline artemis-core-client/src/main/java/org/apache/activemq/artemis/core/remoting/impl/netty/SharedEventLoopGroup.java]: Could it be this that hung?

Comment (jbertram) [inline artemis-core-client/src/main/java/org/apache/activemq/artemis/core/remoting/impl/netty/SharedEventLoopGroup.java]: Possibly. You'll need to run the tests and make a determination.

**Comment by jbertram** [comment]
What's the status on this?

**Comment by zregvart** [comment]
@jbertram sorry didn't find the time to look into this, will take another look towards the middle or end of the week, thank you for your patience 👍

**Comment by zregvart** [comment]
trying to trigger CI

**Comment by zregvart** [comment]
@jbertram I finally got this to work, would you mind having another look? Thanks 👍

**Comment by zregvart** [comment]
Thanks 👍

**Pull request merged**